src/client.ts

```diff
@@ -583,12 +589,24 @@ export class VocdoniSDKClient {
    * @returns {Promise<string>} Resulting election id.
    */
   async createElection(election: UnpublishedElection): Promise<string> {
+    invariant(
+      election.maxCensusSize || election.census.type !== CensusType.CSP,
+      'CSP Census needs a max census size set in the election'
+    );
+
+    const chainId = await this.fetchChainId();
+
     if (!election.census.isPublished) {
       await this.createCensus(election.census as PlainCensus | WeightedCensus);
+    } else if (!election.maxCensusSize) {
+      await this.fetchCensusInfo(election.census.censusId).then((censusInfo) => {
+        election.census.size = censusInfo.size;
+        election.census.weight = censusInfo.weight;
+      });
+    } else if (election.maxCensusSize && election.maxCensusSize > this.chainData.maxCensusSize) {
```

Comment:
@marcvelmer  is there here any value control relevant? Like >0 or any relation with the census.size?

Also not sure if this comment should go here or in types/election/election.ts (if any changes are made here probably they should be included also in the variable setter?)